CppUTestExt tests failing on multiple architectures

Hello, I uploaded version 3.8 to Debian and unfortunately the package did not build on all architectures since the test suite is failing on many of them:
https://buildd.debian.org/status/package.php?p=cpputest

The errors are not the same on all architectures. Those I care the most about are armel and s390x:
https://buildd.debian.org/status/fetch.php?pkg=cpputest&arch=s390x&ver=3.8-2&stamp=1465996058
https://buildd.debian.org/status/fetch.php?pkg=cpputest&arch=armel&ver=3.8-2&stamp=1465996758

On s390x you get this:
```
tests/CppUTestExt/MockSupport_cTest.cpp:194: error: Failure in TEST(MockSupport_c, outputParametersOfType)
	expected <2 0x2>
	but was  <1 0x1>
```
On armel you get this:
```
tests/CppUTestExt/IEEE754PluginTest.cpp:109: error: Failure in TEST(FE__with_Plugin, should_fail_only_once_when_all_flags_are_set)
	expected <1 0x1>
	but was  <0 0x0>

....
tests/CppUTestExt/IEEE754PluginTest.cpp:72: error: Failure in TEST(FE__with_Plugin, should_fail____when__FE_INEXACT____is_set_and_enabled)
./include/CppUTest/TestTestingFixture.h:123: error:
	actual <.
OK (1 tests, 1 ran, 5 checks, 0 ignored, 0 filtered out, 0 ms)

>
	did not contain  <IEEE754_CHECK_CLEAR(FE_INEXACT) failed>

.
tests/CppUTestExt/IEEE754PluginTest.cpp:65: error: Failure in TEST(FE__with_Plugin, should_fail____when__FE_UNDERFLOW__is_set)
./include/CppUTest/TestTestingFixture.h:123: error:
	actual <.
OK (1 tests, 1 ran, 5 checks, 0 ignored, 0 filtered out, 0 ms)

>
	did not contain  <IEEE754_CHECK_CLEAR(FE_UNDERFLOW) failed>

.
tests/CppUTestExt/IEEE754PluginTest.cpp:58: error: Failure in TEST(FE__with_Plugin, should_fail____when__FE_OVERFLOW___is_set)
./include/CppUTest/TestTestingFixture.h:123: error:
	actual <.
OK (1 tests, 1 ran, 5 checks, 0 ignored, 0 filtered out, 0 ms)

>
	did not contain  <IEEE754_CHECK_CLEAR(FE_OVERFLOW) failed>

.
tests/CppUTestExt/IEEE754PluginTest.cpp:51: error: Failure in TEST(FE__with_Plugin, should_fail____when__FE_DIVBYZERO__is_set)
./include/CppUTest/TestTestingFixture.h:123: error:
	actual <.
OK (1 tests, 1 ran, 5 checks, 0 ignored, 0 filtered out, 0 ms)

>
	did not contain  <IEEE754_CHECK_CLEAR(FE_DIVBYZERO) failed>
```

--- a/include/CppUTest/CppUTestConfig.h
+++ b/include/CppUTest/CppUTestConfig.h
@@ -174,7 +174,7 @@
 
 #if CPPUTEST_USE_STD_C_LIB && (!defined(_MSC_VER) || (_MSC_VER >= 1800))
 #define CPPUTEST_HAVE_FENV
-#if defined(__WATCOMC__)
+#if defined(__WATCOMC__) || defined(__ARMEL__) || defined(__m68k__)
 #define CPPUTEST_FENV_IS_WORKING_PROPERLY 0
 #else
 #define CPPUTEST_FENV_IS_WORKING_PROPERLY 1

--- a/tests/CppUTestExt/MockSupport_cTest.cpp
+++ b/tests/CppUTestExt/MockSupport_cTest.cpp
@@ -186,8 +186,8 @@ TEST(MockSupport_c, outputParameters_differentType)
 
 TEST(MockSupport_c, outputParametersOfType)
 {
-    long param = 1;
-    const long retval = 2;
+    int param = 1;
+    const int retval = 2;
     mock_c()->installCopier("typeName", typeCopy);
     mock_c()->expectOneCall("foo")->withOutputParameterOfTypeReturning("typeName", "out", &retval);
     mock_c()->actualCall("foo")->withOutputParameterOfType("typeName", "out", &param);

